lib/bundler/plugin/state_file.rb

```diff
@@ -0,0 +1,46 @@
+require "yaml"
+
+module Bundler
+  module Plugin
+    class StateFile
+
+      # A super basic persitance mechanism for installed plugin. 
+      # As of now, it only stores the require entry points. It could be much more detailed.
+      def initialize
+      	@path = File.expand_path("~/.bundler/plugins.yml")
```

Comment:
Yup, only for global plugins. 
Entry points for project based plugins are in the Gemfile itself. I figured, you don't need another file.